player/lua/console.lua

```diff
@@ -1973,6 +2019,22 @@ end)
 
 mp.register_event('shutdown', function ()
     mp.del_property('user-data/mpv/console')
+
+    if opts.persist_history == false or history_to_save == '' then
+        return
+    end
+
+    local history_path = mp.command_native({'expand-path', opts.history_path})
+    local history_file, error_message = io.open(history_path, 'a')
```

Comment:
Do we care about new-lines portability? Because if so, same as watch history this needs to be saved (and read) as binary to avoid new line conversion.

Reply:
I don't think that matters. Text file is convenient for being easy to edit, I used it to manually remove mistyped commands from the history.

Reply:
If you use `\n` always it will still be easy to edit. It just that Windows and macOS will not convert to `\r\n` or `\r`.